pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -702,9 +713,11 @@ func TestGenericScheduler(t *testing.T) {
 			wErr: &framework.FitError{
 				Pod:         &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-prefilter", UID: types.UID("test-prefilter")}},
 				NumAllNodes: 2,
-				FilteredNodesStatuses: framework.NodeToStatusMap{
-					"1": framework.NewStatus(framework.UnschedulableAndUnresolvable, "injected unschedulable status"),
-					"2": framework.NewStatus(framework.UnschedulableAndUnresolvable, "injected unschedulable status"),
+				Diagnosis: &framework.Diagnosis{
+					NodeToStatusMap: framework.NodeToStatusMap{
+						"1": framework.NewStatus(framework.UnschedulableAndUnresolvable, "injected unschedulable status"),
+						"2": framework.NewStatus(framework.UnschedulableAndUnresolvable, "injected unschedulable status"),
+					},
```

Comment:
It's b/c the status is of type Error, and as I mentioned in https://github.com/kubernetes/kubernetes/pull/98041#issuecomment-765075279, the failedPlugin carried in an Error status won't help in moving pods between queues - Error is interpreted as a fault, not a regular FitError.

Reply:
aren't we building a `FitError` here?

Reply:
Ah, it should embed FailedPlugins. (sorry, misread with another test case)

It can pass is b/c the verifying logic is relaxed without checking the FailedPlugins field:

```go
			if err != test.wErr && !strings.Contains(err.Error(), test.wErr.Error()) {
				t.Errorf("Unexpected error: %v, expected: %v", err.Error(), test.wErr)
			}
```

As we have other tests covering the FailedPlugins part, do you prefer to leave it as is, or complement the wantErr obj and the testing logic?

Reply:
I prefer we have the FailedPlugins part so that eventually we can move to cmp

Reply:
Fair enough. Let me complement that.

Reply:
Updated. To ensure correctness, I introduced `reflect.DeepEqual`. I added another TODO in #94696.

Reply:
leave a TODO as comment in the code too.

Reply:
Done.